File: frontend/src/app/services/booking-service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Booking } from '../models/booking';
import { Observable } from 'rxjs/internal/Observable';

@Injectable({
  providedIn: 'root',
})
export class BookingService {
  public baseUrl = "";
  constructor(private httpClient : HttpClient) { }

  addBooking(booking : Booking,movieId:number,userId:number) : Observable<Booking> {
     return this.httpClient.post<Booking>(this.baseUrl + "/" + movieId + "/" + userId, booking);
  }

  getUserBookings(userId : number) : Observable<Booking[]> {
    return this.httpClient.get<Booking[]>(this.baseUrl + "/user/" + userId);
  }

  updateBooking(booking : Booking) : Observable<Booking> {
    return this.httpClient.put<Booking>(this.baseUrl + "/" + booking.bookingId , booking);
  }

  deleteBooking(id : number) : Observable<boolean>{
    return this.httpClient.delete<boolean>(this.baseUrl + "/" + id);
  }

  getAllBookings() : Observable<Booking[]>{
    return this.httpClient.get<Booking[]>(this.baseUrl);
  }

}

File: backend/src/main/java/com/example/backend/Service/BookingService.java
package com.example.backend.Service;

import java.util.List;

import com.example.backend.Entity.Booking;
import com.example.backend.Entity.BookingDto;

public interface BookingService {
     Booking createBooking(BookingDto bookingDto,long movieId, int userId);
     Booking getBookingById(long bookingId);
     List<Booking> getAllBookings();
     void deleteBooking(long bookingId);
     List<Booking> findBookingsByUser(int userId);
     Booking updateBooking(long bookingId, BookingDto bookingDto);
}
